cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -71,6 +77,103 @@ public Schema getSchema() {
     return (T) fields.get(fieldName);
   }
 
+  /**
+   * Get {@link LocalDate} from field of logical type date.
+   * @param fieldName date field to get.
+   * @return value of the field as an object of {@link LocalDate}, returns null if given field is not of
+   * logical type date
+   */
+  public LocalDate getDate(String fieldName) {
+    if (!isValidLogicalType(fieldName, Schema.LogicalType.DATE)) {
+      return null;
+    }
+
+    return LocalDate.ofEpochDay(((Integer) fields.get(fieldName)).longValue());
```

Comment:
`isValidLogicalType` has a check to return null in case value is null, but I will add a test with null values as well:

```java
 if (fields.get(fieldName) == null) {
      return false;
 }
```